tagmanager/src/tm_source_file.c

```diff
@@ -223,20 +245,36 @@ static void tm_source_file_destroy(TMSourceFile *source_file)
 	source_file->tags_array = NULL;
 }
 
-/** Frees a TMSourceFile structure, including all contents. Before calling this
- function the TMSourceFile has to be removed from the TMWorkspace. 
- @param source_file The source file to free.
+/** Decrements the reference count of @a source_file
+ *
+ * If the reference count drops to 0, then @a source_file is freed, including all contents.
+ * Make sure the @a source_file is already removed from any TMWorkSpace before the
+ * this happens.
+ * @param source_file The source file to free.
+ * @see tm_workspace_remove_source_file()
 */
 GEANY_API_SYMBOL
 void tm_source_file_free(TMSourceFile *source_file)
 {
-	if (NULL != source_file)
```

Comment:
IMHO if its in the API for plugins to use its probably better to be conservative and protect against obvious errors like nulls.  

Things like the `g_return*` series and asserts that can be compiled out are fine for use within an application where there is only a finite usage,and if there are no noisy messages or blow ups after a reasonable test, then its reasonably safe to remove them.

But a plugin API can have a new piece of code using it at any time, so the safety checks should stay.  In an ideal world the whole plugin API should be audited for crash safety, but unless somebody volunteers, then lets just add safety checks as we are changing something in the area and notice it.